### PR TITLE
[refactor_Image] refactor Image done

### PR DESCRIPTION
**Pull request by kolesnikdi**
Зламав твою логіку збереження файлів

### CODE REVIEW

File: company/serializers.py [hunk 1]
```diff
@@ -20,33 +23,50 @@ class Meta:
...
-            legal_address=Address.objects.create(**legal_address),
-            actual_address=Address.objects.create(**actual_address),
-            **validated_data,
-        )
+        legal_address_data = validated_data.pop('legal_address')
+        actual_address_data = validated_data.pop('actual_address')
+        logo_data = validated_data.pop('logo')
+        with transaction.atomic():  # atomic should stop other requests to db until we make the changes
+            """ Works only with custom ImageSerializer.
+            Does not work directly through the serializers.ImageField."""
```
Comment (ArtemKhriapa) [inline company/serializers.py:53]: is this comment necessary?

File: company/serializers.py [hunk 2]
```diff
@@ -20,33 +23,50 @@ class Meta:
...
+        legal_address_data = validated_data.pop('legal_address')
+        actual_address_data = validated_data.pop('actual_address')
+        logo_data = validated_data.pop('logo')
         # rewrite the address separately by using 'Serializer'
-        instance.legal_address = AddressSerializer().update(instance.legal_address, legal_address)
-        instance.actual_address = AddressSerializer().update(instance.actual_address, actual_address)
+        instance.legal_address = AddressSerializer().update(instance.legal_address, legal_address_data)
+        instance.actual_address = AddressSerializer().update(instance.actual_address, actual_address_data)
+        instance.logo = ImageSerializer().update(instance.logo, logo_data)
         return super().update(instance, validated_data)  # rewrite company by default method - 'update'
```
Comment (ArtemKhriapa) [inline company/serializers.py:70]: is this comment still necessary? is it valid?

File: image/business_logic.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+"""Path to upload image for different apps request"""
```
Comment (ArtemKhriapa) [inline image/business_logic.py:1]: move this comment inside function

File: media/company_1/company_logo/5_Valid_logo.jpg
Comment (ArtemKhriapa) [inline media/company_1/company_logo/5_Valid_logo.jpg:1]: cut `media` from commit

Comment (ArtemKhriapa) [reply media/company_1/company_logo/5_Valid_logo.jpg:1]: add `media` folder into gitignore

File: company/serializers.py [hunk 4]
```diff
@@ -20,33 +23,50 @@ class Meta:
...
+            )
         return company
 
     def update(self, instance, validated_data):
-        legal_address = validated_data.pop('legal_address')
-        actual_address = validated_data.pop('actual_address')
+        legal_address_data = validated_data.pop('legal_address')
+        actual_address_data = validated_data.pop('actual_address')
+        logo_data = validated_data.pop('logo')
         # rewrite the address separately by using 'Serializer'
```
[the remark this answers is not in the document]
Comment (kolesnikdi) [reply company/serializers.py:66]: done

File: company/serializers.py (hunk 2 above)
Comment (kolesnikdi) [reply company/serializers.py:70]: done

File: image/business_logic.py (hunk 3 above)
Comment (kolesnikdi) [reply image/business_logic.py:1]: done

**Review by ArtemKhriapa — APPROVED** [review]

**Pull request merged**